src/compas/robots/base_artist/_artist.py

```diff
@@ -0,0 +1,309 @@
+        transformation from the robot model.
+
+        Parameters
+        ----------
+        link : :class:`compas.robots.Link`, optional
+            Link instance to create. Defaults to the robot model's root.
+        name : :obj:`str`, optional
+            Subdomain identifier. # !!! there's got to be a better explanation.
+
+        Returns
+        -------
+        None
+        """
+        if link is None:
+            link = self.model.root
+
+        for item in itertools.chain(link.visual, link.collision):
+            meshes = self.model._get_item_meshes(item)  # !!! this is weird
+
+            if meshes:
+                is_visual = hasattr(item, 'get_color')
+                color = item.get_color() if is_visual else None
+
+                native_geometry = []
+                for i, mesh in enumerate(meshes):
+                    # create native geometry
+                    mesh_type = 'visual' if is_visual else 'collision'
+                    mesh_name_components = [self.model.name, mesh_type, link.name, str(i)]
+                    if name:
+                        mesh_name_components.insert(2, name)
+                    mesh_name = '.'.join(mesh_name_components)
+                    native_mesh = self.draw_geometry(mesh, name=mesh_name, color=color)
+                    # transform native geometry based on saved init transform
+                    self.transform(native_mesh, item.init_transformation)
+                    # append to list
+                    native_geometry.append(native_mesh)
+
+                item.native_geometry = native_geometry
+                item.current_transformation = Transformation()
+
+        for child_joint in link.joints:
+            self.create(child_joint.child_link)
+
+    def scale(self, factor):
+        """Scales the robot model's geometry by factor (absolute).
+
+        Parameters
+        ----------
+        factor : float
+            The factor to scale the robot with.
+
+        Returns
+        -------
+        None
+        """
+        self.model.scale(factor)  # scale the model
+
+        relative_factor = factor / self.scale_factor  # relative scaling factor
+        transformation = Scale.from_factors([relative_factor] * 3)
+        self.scale_link(self.model.root, transformation)
+        self.scale_factor = factor
+
+    def scale_link(self, link, transformation):
+        """Recursive function to apply the scale transformation on each link.
+        """
+        for item in itertools.chain(link.visual, link.collision):
+            # Some links have only collision geometry, not visual. These meshes
+            # have not been loaded.
+            if item.native_geometry:
+                for geometry in item.native_geometry:
+                    self.transform(geometry, transformation)
+
+        for child_joint in link.joints:
+            # Recursive call
+            self.scale_link(child_joint.child_link, transformation)
```

Comment:
Should an attached_tool_model also be scaled?

Reply:
agree, it should be scaled.